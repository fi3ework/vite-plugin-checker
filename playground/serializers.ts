import os from 'os'

const winNewLineReg = /\/r\/n/gim
const winSepReg = /\\/g

function doesUseDoubleSlashAsPath(val: string) {
  return val.includes('//vite-plugin-checker//')
}

export const normalizeWindowsLogSerializer = {
  print(val: string, serialize) {
    let result = val
    if (os.platform() === 'win32') {
      result = result.replaceAll(winNewLineReg, '/n')
      result = result.replaceAll(process.cwd().replace(winSepReg, '/'), '<PROJECT_ROOT>')

      if (doesUseDoubleSlashAsPath(result)) {
        result = result.replaceAll(
          `//a//vite-plugin-checker//vite-plugin-checker//playground-temp`,
          '<PROJECT_ROOT>/playground-temp'
        )
        result = result.split('//').join('/')
      }
    }

    return serialize(result)
  },
  test(val) {
    if (typeof val !== 'string') return false

    if (
      (os.platform() === 'win32' &&
        (val.includes(process.cwd().replace(winSepReg, '/')) || winNewLineReg.test(val))) ||
      doesUseDoubleSlashAsPath(val)
    ) {
      return true
    }

    return false
  },
}
