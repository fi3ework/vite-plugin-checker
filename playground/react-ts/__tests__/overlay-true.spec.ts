import {
  getHmrOverlay,
  getHmrOverlayText,
  killServer,
  pollingUntil,
  preTest,
  sleepForEdit,
  viteServe,
} from 'vite-plugin-checker/__tests__/e2e/Sandbox/Sandbox'
import {
  editFile,
  sleep,
  testDir,
  WORKER_CLEAN_TIMEOUT,
} from 'vite-plugin-checker/__tests__/e2e/testUtils'

import { copyCode } from '../../../scripts/jestSetupFilesAfterEnv'
import { serializers } from '../../../scripts/serializers'

expect.addSnapshotSerializer(serializers)

beforeAll(async () => {
  await preTest()
})

afterAll(async () => {
  await sleep(WORKER_CLEAN_TIMEOUT)
})

describe('overlay', () => {
  beforeEach(async () => {
    await copyCode()
  })

  afterEach(async () => {
    await killServer()
  })

  it('overlay prompts and changes', async () => {
    await viteServe({ cwd: testDir, launchPage: true })
    await pollingUntil(getHmrOverlay, (dom) => !!dom)
    const [message1, file1, frame1] = await getHmrOverlayText()
    expect(message1).toMatchSnapshot()
    expect(file1).toMatchSnapshot()
    expect(frame1).toMatchSnapshot()

    console.log('-- overlay update after edit --')
    editFile('src/App.tsx', (code) => code.replace('useState<string>(1)', 'useState<string>(2)'))
    await sleepForEdit()
    await pollingUntil(getHmrOverlay, (dom) => !!dom)
    const [, , frame2] = await getHmrOverlayText()
    expect(frame2).toMatchSnapshot()

    console.log('-- overlay dismiss after fix error --')
    editFile('src/App.tsx', (code) => code.replace('useState<string>(2)', `useState<string>('x')`))
    await sleep(6000)
    await expect(getHmrOverlayText()).rejects.toThrow(
      'Invariant failed: <vite-plugin-checker-error-overlay> shadow dom is expected to be found, but got null'
    )
  })
})
