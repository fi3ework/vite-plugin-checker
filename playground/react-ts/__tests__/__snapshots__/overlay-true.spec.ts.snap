// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`overlay overlay prompts and changes 1`] = `"Argument of type 'number' is not assignable to parameter of type 'string | (() => string)'."`;

exports[`overlay overlay prompts and changes 2`] = `"<PROJECT_ROOT>/temp/react-ts/src/App.tsx:6:46"`;

exports[`overlay overlay prompts and changes 3`] = `
"    4 |
    5 | function App() {
  > 6 |   const [count, setCount] = useState<string>(1)
      |                                              ^
    7 |   return (
    8 |     <div className=\\"App\\">
    9 |       <header className=\\"App-header\\">"
`;

exports[`overlay overlay prompts and changes 4`] = `
"    4 |
    5 | function App() {
  > 6 |   const [count, setCount] = useState<string>(2)
      |                                              ^
    7 |   return (
    8 |     <div className=\\"App\\">
    9 |       <header className=\\"App-header\\">"
`;
